src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -266,7 +266,7 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr curlPtr)
                 catch (Exception exc)
                 {
                     EventSourceTrace("Unexpected exception: {0}", exc, easy: easy);
-                    easy.FailRequest(exc);
+                    easy.FailRequest(CreateHttpRequestException(exc));
```

Comment:
> I would think we want the behavior in .NET Core to perhaps have HttpRequestException as outer and then inside that is a handler exception (WInHttpException or CurlException) and inside that is the exception from the user-provided callback

Thanks, but what's the value of the intermediary exception?  You're trying to keep the number of exceptions in the hierarchy the same?